File: ChessGameRCdb/Logic/Interfaces/IFigure.cs
using System.Collections.Generic;

namespace ChessGame.Logic
{
    public interface IFigure
    {
        Color Color { get; }
        Coordinate Coordinate { get; }
        FigureType FigureType { get; }
        IEnumerable<IMove> PossibleMoves { get; }
        IEnumerable<MoveOption> MoveOptions(IBoard board);
        bool IsMoveAllowed(IBoard currentBoard, Coordinate endPoint);
        bool IsPlayersFigure(Color player, Coordinate endPoint);
        bool IsEnemysFigure(Color player, Coordinate endPoint);
        bool IsInPosition(Coordinate endPoint);
        void SetPosition(Coordinate endPoint);
        bool IsAttackingOpponentsKingOnPlayersMove(IBoard board);
    }
}

File: ChessGameRCdb/ClientAppBoard/src/board/interface/IFigure.ts
import { FigureType } from 'board/enum/FigureType'
import { Player } from 'board/enum/Player'
import ISquare from 'board/interface/ISquare'

export default interface IFigure {
    Id: number
    Type: FigureType
    Player: Player,
    Square: ISquare
    EnableMoves: Array<ISquare> | undefined
}